From 23ccb99841bee49014176c37a894dd653429a3a8 Mon Sep 17 00:00:00 2001
From: Chih-Hung Hsieh <chh@google.com>
Date: Thu, 2 Nov 2017 10:24:36 -0700
Subject: [PATCH 16/67] Use -Werror in packages/apps/Gallery2

* Keep constant-conversion warnings to fix later.

Bug: 66996870
Test: build with WITH_TIDY=1
Change-Id: Ib93e954e025fd2f0202960df20583b50b97fce7f
---
 jni/Android.mk            | 11 ++++++++++-
 jni_jpegstream/Android.mk |  1 +
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/jni/Android.mk b/jni/Android.mk
index 1a45a90be..28c6d78b7 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -1,8 +1,16 @@
 LOCAL_PATH:= $(call my-dir)
 
+Gallery2_jni_cflags := \
+    -Wall -Wextra -Werror \
+    -Wno-error=constant-conversion \
+    -Wno-unused-parameter \
+
+# to fix implicit conversion from 'int' to 'char', (255 to -1, 128 to -128)
+
 include $(CLEAR_VARS)
 
 LOCAL_CFLAGS += -DEGL_EGLEXT_PROTOTYPES
+LOCAL_CFLAGS += $(Gallery2_jni_cflags)
 
 LOCAL_SRC_FILES := jni_egl_fence.cpp
 
@@ -44,7 +52,8 @@ LOCAL_SRC_FILES := filters/gradient.c \
                    filters/tinyplanet.cc \
                    filters/kmeans.cc
 
-LOCAL_CFLAGS    += -ffast-math -O3 -funroll-loops
+LOCAL_CFLAGS += -ffast-math -O3 -funroll-loops
+LOCAL_CFLAGS += $(Gallery2_jni_cflags)
 LOCAL_LDLIBS := -llog -ljnigraphics
 LOCAL_ARM_MODE := arm
 
diff --git a/jni_jpegstream/Android.mk b/jni_jpegstream/Android.mk
index 40b9bd53d..76946f758 100644
--- a/jni_jpegstream/Android.mk
+++ b/jni_jpegstream/Android.mk
@@ -18,6 +18,7 @@ LOCAL_SDK_VERSION   := 17
 LOCAL_ARM_MODE := arm
 
 LOCAL_CFLAGS    += -ffast-math -O3 -funroll-loops
+LOCAL_CFLAGS += -Wall -Wextra -Werror
 LOCAL_CPPFLAGS += $(JNI_CFLAGS)
 LOCAL_LDLIBS := -llog
 
-- 
2.17.1

